lib/npm.js

```diff
@@ -295,6 +295,13 @@
           umask: umask
         }
 
+        if (config.get('proxy') && !process.env['HTTP_PROXY']){
+          process.env['HTTP_PROXY'] = config.get('proxy')
```

Comment:
I'm not entirely comfortable with tweaking the env globally like this. Particularly because this kind of thing makes refactoring to put config all in one place much harder. (And that's something we desperately need to do.)

I'd prefer that this either get added separately to `lib/utils/lifecycle.js` and `lib/utils/git.js` or put into a library that both those modules use.

Either way, it will need a test and a documentation update.

Docs would be `doc/misc/npm-scripts.md` and `doc/misc/npm-config.md` (where the proxy config options are discussed).